### DISCUSSION TITLE
Docker project setup

### DISCUSSION THREAD

**Discussion by officialhumanofhumidity** [chatwoot/chatwoot#8000] [Q&A]
### Describe the bug

I am trying to setup this project on my local machine using the docker installation process in which i encountered an error when i was trying to do docker compose build so the error was for the github repo clone which is happening for production  env which should not happen in develpement env.
![Screenshot from 2023-09-21 20 20 22](https://github.com/chatwoot/chatwoot/assets/60418079/15709841-b699-484f-9203-d69111c005d4)

### To Reproduce

https://www.chatwoot.com/docs/contributing-guide/environment-setup/docker follow this step in step 3 i encountered the issue.

### Environment

Other [please specify in the description]

### Cloud Provider

Azure

### Platform

Browser

### Operating system

elementry os 

### Docker (if applicable)

docker version
Docker version 24.0.5, build 24.0.5-0ubuntu1~22.04.1

docker info 
Client:
 Version:    24.0.5
 Context:    default
 Debug Mode: false

Server:
 Containers: 5
  Running: 0
  Paused: 0
  Stopped: 5
 Images: 70
 Server Version: 24.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 
 runc version: 
 init version: 
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.2.0-33-generic
 Operating System: elementary OS 7.1 Horus
 OSType: linux
 Architecture: x86_64
 CPUs: 8
 Total Memory: 7.098GiB
 Name: HP-Laptop-14s-fq1xxx-5f6d6e69
 ID: 931bd4cf-4575-41f0-91f8-e330e2057831
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

docker compose version 
docker-compose version 1.29.2, build unknown
docker-py version: 5.0.3
CPython version: 3.10.12
OpenSSL version: OpenSSL 3.0.2 15 Mar 2022

**Comment by pranavrajs** [comment]
@officialhumanofhumidity This seems like an issue with the network. @vishnu-narayanan Can you please take a look?

**Comment by officialhumanofhumidity** [comment]
hey  @vishnu-narayanan, please go through the issue mentioned above and help me set up the project.

**Comment by joseEkcit** [comment]
@officialhumanofhumidity These "Could not resolve host: github.com" errors are usually solved by just restarting the docker service and daemon.

**Comment by vishnu-narayanan** [comment]
@officialhumanofhumidity The error you have shared is related to network and could be temporary. Could you try again?  Restart the VM or maybe try on another instance.

**Comment by officialhumanofhumidity** [comment]
hey @vishnu-narayanan  i even tried installing in different machine now still facing the same issue 



the error am getting is build failed near line no 46
file:46
--------------------
  45 |     # Do not install development or test gems in production
  46 | >>> RUN if [ "$RAILS_ENV" = "production" ]; then \
  47 | >>>   bundle config set without 'development test'; bundle install -j 4 -r 3; \
  48 | >>>   else bundle install -j 4 -r 3; \
  49 | >>>   fi
  50 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c if [ \"$RAILS_ENV\" = \"production\" ]; then   bundle config set without 'development test'; bundle install -j 4 -r 3;   else bundle install -j 4 -r 3;   fi" did not complete successfully: exit code: 11
ERROR: Service 'base' failed to build : Build failed





and the docker file having this below line of scripts from line no 46

i also tried changing the rails_env from production to development even that didn't worked.

**Comment by vishnu-narayanan** [comment]
@officialhumanofhumidity Looks like it is now failing at installing rails dependencies stage. There is not enough information to debug here. What is the instance/machine spec?

**Comment by officialhumanofhumidity** [comment]
@vishnu-narayanan 

system specs 
System:
  Kernel: 6.2.0-32-generic arch: x86_64 bits: 64 compiler: N/A Desktop: GNOME
    v: 44.3 Distro: Ubuntu 23.04 (Lunar Lobster)

Network:
  Device-1: Intel Wireless 8265 / 8275 driver: iwlwifi v: kernel
    bus-ID: 01:00.0
  IF: wlp1s0 state: up mac: <filter>
  IF-ID-1: br-b19f5981adb3 state: down mac: <filter>
  IF-ID-2: docker0 state: down mac: <filter>
  IF-ID-3: gpd0 state: down mac: N/A

**Comment by vishnu-narayanan** [comment]
@officialhumanofhumidity CPU/RAM?

**Comment by officialhumanofhumidity** [comment]
processor AMD Ryzen™ 5 3500U with Radeon™ Vega Mobile Gfx × 8

ram 8.0 GB

**Reply by vishnu-narayanan** [reply]
@officialhumanofhumidity Could you try this on an instance with more RAM?

**Comment by pranavrajs** [comment]
Moving this to discussions as it is still not clear if this is a bug.
